app/commands/resumes/update.rb

```diff
@@ -0,0 +1,34 @@
+module Resumes
+  class Update
+    attr_accessor :user, :attributes, :current_resume_id
+
+    def initialize(user:, params:)
+      @user = user
+      @attributes = resume_attributes(params)
+      @current_resume_id = params[:current_resume_id]
+    end
+
+    def call!
+      ActiveRecord::Base.transaction do
+        user.current_resume.update!(current: false)
+
+        if attributes[:resume]
+          user.resumes.create!(attributes)
+          next
+        end
+
+        user.resumes.find_by(id: current_resume_id).update!(current: true)
+      end
```

Comment:
I think we can optimize this quickly a tiny bit. We should be able to `next` if `attributes[:resume].blank?` && current_resume.id == current_resume id.

Other than that this makes sense what is happening. Might be worth adding a small comment explaining that they can chose a different resume to make as current (took me a min to figure that out), and not need to upload one.

Reply:
I think some comments might make sense to clarify some, but I think the logic flow may be confusing, not least of which because `next` is not really a correct word for the context, but rubocop is irritating (Rails/TransactionExitStatement). I can update to ignore rubocop in this context.

It should be this:
```ruby
  ActiveRecord::Base.transaction do
    user.current_resume.update!(current: false)
    return user.resumes.create!(attributes) if attributes[:resume]
    user.resumes.find_by(id: current_resume_id).update!(current: true)
  end
```

Reply:
what rubocop is saying is correct. I understand the `next` and `return` difference. It was more asking if we could include a guard clause because `user.current_resume.update!(current: false)` and `user.resumes.find_by(id: current_resume_id).update!(current: true)` might be the exact same record which could already be true.

Reply:
gotcha. updated.